### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by h-vetinari** [comment]
Nice! 👍 

Is this a good time to absorb gcc/gxx outputs like you [suggested](https://github.com/conda-forge/clang-compiler-activation-feedstock/issues/118#issuecomment-1998660198)?

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -44,10 +47,13 @@ _libgcc_mutex:
 use_system_gcc:
   - True         # [not (linux64 or aarch64 or ppc64le or s390x)]
   - False        # [linux64 or aarch64 or ppc64le or s390x]
+channel_sources:
+  - conda-forge/label/sysroot-with-crypt,conda-forge
```
Comment (beckermr) [inline recipe/conda_build_config.yaml:51]: For my own knowledge, why do we need the sysroot with crypt here?

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (isuruf) [reply recipe/conda_build_config.yaml:51]: `gcc<14`'s libsanitizer compiles against crypt.h and links to libcrypt.a. It's hard to get this package to use our `libxcrypt` package and also creates bootstrapping problems.

**Pull request merged**